quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/jaxrs/BadRequestFilter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.quarkus.runtime.integration.jaxrs;
+
+import java.io.IOException;
+
+import org.jboss.logging.Logger;
+import org.keycloak.models.KeycloakSession;
+
+import jakarta.annotation.Priority;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.BadRequestException;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.container.PreMatching;
+import jakarta.ws.rs.ext.Provider;
+
+@Provider
+@PreMatching
+@Priority(1)
+public class BadRequestFilter implements ContainerRequestFilter {
```

Comment:
I think it would be worth splitting this into a separate PR. It kinda opens a whole new discussion since we have startup basically saying to to `http://0.0.0.0` ;)

If I remember correctly the SRE team also use this as a work-around due to Gatling not dealing with cookies correctly. @ahus1 FIY

Reply:
https://github.com/keycloak/keycloak/issues/33785